src/wasm/mod.rs

```diff
@@ -0,0 +1,80 @@
+use crate::utils;
+use crate::Handler;
+
+use std::ops::Add;
+
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct Server {
+    handler: Handler,
+}
+
+#[wasm_bindgen]
+pub struct ServerResponse {
+    message: Option<String>,
+    error: Option<String>,
+}
+
+#[wasm_bindgen]
+impl ServerResponse {
+    pub fn get_message(&self) -> Option<String> {
+        self.message.clone()
+    }
+
+    pub fn get_error(&self) -> Option<String> {
+        self.error.clone()
+    }
+}
+
+impl Default for Server {
```

Comment:
Dumb rust question from having just used `Default` this weekend: why is this the same as the `new` definition on line 41?

Reply:
this is so structs that have a Server in them can derive default as this struct has a default